test/types/schema.test.ts

```diff
@@ -2086,3 +2086,31 @@ function gh15751() {
   const doc = new TestModel();
   expectType<Types.ObjectId>(doc.myId);
 }
+
+function testNewSchemaWithMethodsAndVirtuals() {
+  const schema = new Schema(
+    { name: String },
+    {
+      virtuals: {
+        upperName: {
+          get(): string | undefined {
```

Comment:
I worked on this issue a little bit when I commented on it, the root cause to rigger the issue was that OP was using `this: any` which somehow caused the document to not know about the methods and virtuals.
Worth testing with this before merging, or maybe even adding a test explicitly with `this: any`.

Reply:
@AbdelrahmanHafez `this: any` works fine with the changes suggested in this PR, following screenshot shows tests passing:

<img width="1470" height="730" alt="image" src="https://github.com/user-attachments/assets/bfe662f1-f0ff-4e40-b150-fd4e11b85c18" />